core/variant_op.cpp

```diff
@@ -2489,110 +2380,14 @@ Variant Variant::get(const Variant& p_index, bool *r_valid) const {
-				const Array *arr=reinterpret_cast<const Array* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return (*arr)[index];
-				}
-			}
-
-		} break;
-		case RAW_ARRAY: {
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<uint8_t> *arr=reinterpret_cast<const DVector<uint8_t>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-
-		} break;
-		case INT_ARRAY: {
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<int> *arr=reinterpret_cast<const DVector<int>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-		} break;
-		case REAL_ARRAY: {
-
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<real_t> *arr=reinterpret_cast<const DVector<real_t>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-
-		} break;
-		case STRING_ARRAY: {
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<String> *arr=reinterpret_cast<const DVector<String>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-
-		} break;	//25
-		case VECTOR2_ARRAY: {
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<Vector2> *arr=reinterpret_cast<const DVector<Vector2>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-
-		} break;
-		case VECTOR3_ARRAY: {
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<Vector3> *arr=reinterpret_cast<const DVector<Vector3>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-
-		} break;
-		case COLOR_ARRAY: {
-
-			if (p_index.get_type()==Variant::INT || p_index.get_type()==Variant::REAL) {
-
-				int index = p_index;
-				const DVector<Color> *arr=reinterpret_cast<const DVector<Color>* >(_data._mem);
-
-				if (index >=0 && index <arr->size()) {
-					valid=true;
-					return arr->get(index);
-				}
-			}
-		} break;
+		DEFAULT_OP_ARRAY_CMD(ARRAY, const Array, 0, return (*arr)[index])
+		DEFAULT_OP_DVECTOR_GET(RAW_ARRAY, uint8_t)
+		DEFAULT_OP_DVECTOR_GET(INT_ARRAY, int)
+		DEFAULT_OP_DVECTOR_GET(REAL_ARRAY, real_t)
+		DEFAULT_OP_DVECTOR_GET(STRING_ARRAY, String)
+		DEFAULT_OP_DVECTOR_GET(VECTOR2_ARRAY, Vector2)
+		DEFAULT_OP_DVECTOR_GET(VECTOR3_ARRAY, Vector3)
+		DEFAULT_OP_DVECTOR_GET(COLOR_ARRAY, Color)
 		default: return Variant();
 	}
```

Comment:
AFAIK, in most of the other code #defines are undefined after they are used. :smile: